framework/src/utils/FunctionParserUtils.C

```diff
@@ -81,7 +81,7 @@ FunctionParserUtils::evaluate(ADFunctionPtr & parser)
     return 0.0;
 
   // evaluate expression
-  Real result = parser->Eval(&_func_params[0]);
+  Real result = parser->Eval(_func_params.data());
```

Comment:
Are you relying on `std::vector::data()` to return `nullptr` when the `vector` is empty?

Because [I don't think you can rely on that](https://stackoverflow.com/questions/25419851/what-should-stdvectordata-return-if-the-vector-is-empty).